lib/src/util/class_names.dart

```diff
@@ -121,37 +150,70 @@ class ClassNameBuilder {
+  /// Returns a String representation of the built className, which includes any added classes,
+  /// and none of the blacklisted classes.
   ///
-  /// Duplicate classes will be added.
+  /// Does not check for / remove duplicate CSS classes.
+  ///
+  /// __Necessary to convert a `ClassNameBuilder` instance into a valid [CssClassPropsMixin.className] value:__
+  ///
+  ///     @override
+  ///     render() {
+  ///       var classes = forwardingClassNameBuilder()
+  ///         ..add('foo')
+  ///         ..add('foo--is-active', state.isActive)
+  ///         ..add('foo--is-disabled', state.isDisabled);
+  ///
+  ///       return (Dom.div()
+  ///         ..addProps(copyUnconsumedDomProps())
+  ///         ..className = classes.toClassName()
+  ///       )(props.children);
+  ///     }
+  ///
+  /// > Related: [toClassNameBlacklist]
   String toClassName() {
     String className = _classNamesBuffer.toString();
 
     if (_blacklistBuffer != null && _blacklistBuffer.isNotEmpty) {
       List blacklistedClasses = splitSpaceDelimitedString(_blacklistBuffer.toString());
 
       className = splitSpaceDelimitedString(className)
-        .where((String cssClass) => !blacklistedClasses.contains(cssClass))
-        .join(' ');
+          .where((String cssClass) => !blacklistedClasses.contains(cssClass))
+          .join(' ');
     }
 
     return className;
   }
 
   /// Returns a String representation of only the blacklisted classes.
-  /// Useful for blacklist forwarding.
   ///
-  /// Duplicate classes will be added.
+  /// Does not check for / remove duplicate CSS classes.
+  ///
+  /// __Useful for blacklist forwarding:__
+  ///
+  ///     @override
+  ///     render() {
+  ///       var classes = forwardingClassNameBuilder()
+  ///         ..blacklist('some-class-found-in-the-forwarding-builder', state.becauseReasons);
+  ///
+  ///       return (NestedComponent()
+  ///         ..addProps(copyUnconsumedProps())
+  ///         ..className = classes.toClassName()
+  ///         ..classNameBlacklist = classes.toClassNameBlacklist()
+  ///       )(props.children);
+  ///     }
+  ///
+  /// > Related: [toClassName]
   String toClassNameBlacklist() {
     return _blacklistBuffer == null || _blacklistBuffer.isEmpty
-      ? null
-      : _blacklistBuffer.toString();
+        ? null
+        : _blacklistBuffer.toString();
   }
 
-  /// Returns a Map with the [CssClassProps.className] and [CssClassProps.classNameBlackList] props
+  /// Returns a Map with the [CssClassPropsMixin.className] and [CssClassPropsMixin.classNameBlacklist] props
   /// populated from the return values of [toClassName] and [toClassNameBlacklist], respectively.
   ///
-  /// This method, along with [addFromProps], is useful for merging sets of className/blacklist props.
+  /// > This method, along with [addFromProps], are useful for merging sets of className/blacklist props.
```

Comment:
I believe "is" is correct here, since "along with [addFromProps]" is a separate clause and does not affect the subject of the sentence.

Alternatively, "This method and [addFromProps] are useful" would also be correct.